lib/cocoapods/installer.rb

```diff
@@ -350,7 +350,7 @@ def lock_pod_sources
     def determine_dependency_product_types
       aggregate_targets.each do |aggregate_target|
         aggregate_target.pod_targets.each do |pod_target|
-          pod_target.host_requires_frameworks = aggregate_target.requires_frameworks?
+          pod_target.host_requires_frameworks ||= aggregate_target.requires_frameworks?
```

Comment:
This is debatable, whether that should be a separate criteria or not. I would address this in a separate PR.